src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -130,6 +138,72 @@ function createModel(program: Program): any {
     if (authentication) {
         auth = processServiceAuthentication(authentication);
     }
+    const consumes = getConsumes(program, serviceNamespaceType);
+    let contentTypeParameter = undefined;
+    if (consumes && consumes.length > 0) {
+        contentTypeParameter = {
+            Name: "contentType",
+            NameInRequest: "content-type",
+            Type: {
+                Name: "String",
+                Kind: InputTypeKind.String,
+                IsNullable: false
+            } as InputPrimitiveType,
+            Location: RequestLocation.Header,
+            IsApiVersion: false,
+            IsResourceParameter: false,
+            IsContentType: true,
+            IsRequired: true,
+            IsEndpoint: false,
+            SkipUrlEncoding: false,
+            Explode: false,
+            Kind: InputOperationParameterKind.Constant,
+            DefaultValue:
+                consumes.length === 1
```

Comment:
Can we extract this into a helper function? It looks like logic is the same for `produces` and `consumes`

Reply:
Thanks @AlexanderSher . Good Suggestion. I wrapped them into a function.